Restore coin-config for legacy autotools consumers, including Coin-framework.
The CMake template emits a bare library directory as --libs and omits the
include path; framework builds do not install the helper at all. Supply
usable compile/link flags and retain the absolute framework install name
needed by consumers that do not set an rpath.

--- src/CMakeLists.txt.orig
+++ src/CMakeLists.txt
@@ -250,7 +250,7 @@
     )
     # adapt install directory to allow distribution of dylibs/frameworks in users
     # framework/application bundle
-    set_target_properties(${PROJECT_NAME} PROPERTIES INSTALL_NAME_DIR "@rpath")
+    set_target_properties(${PROJECT_NAME} PROPERTIES INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}")
     if(NOT CMAKE_SKIP_BUILD_RPATH)
       if(CMAKE_VERSION VERSION_LESS 3.9)
         set_target_properties(${PROJECT_NAME} PROPERTIES BUILD_WITH_INSTALL_RPATH TRUE)
@@ -407,16 +407,25 @@
 if(COIN_BUILD_MAC_FRAMEWORK)
   set(COIN_CONFIG_PACKAGE_LOCATION ${COIN_MAC_FRAMEWORK_NAME}.framework/Resources/CMake)
   string(TOLOWER ${COIN_MAC_FRAMEWORK_NAME} COIN_CONFIG_PACKAGE_BASENAME)
+  set(COIN_CONFIG_INCLUDEDIR "${COIN_PKGCFG_FRAMEWORK_DIR}/Resources/include")
+  set(COIN_CONFIG_FRAMEWORKDIR "${COIN_PKGCFG_FRAMEWORK_DIR}")
+  set(COIN_CONFIG_LDFLAGS "")
 else()
   set(COIN_CONFIG_PACKAGE_LOCATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}-${PROJECT_VERSION})
   string(TOLOWER ${PROJECT_NAME} COIN_CONFIG_PACKAGE_BASENAME)
+  set(COIN_CONFIG_INCLUDEDIR "${CMAKE_INSTALL_FULL_INCLUDEDIR}")
+  set(COIN_CONFIG_FRAMEWORKDIR "")
+  set(COIN_CONFIG_LDFLAGS "-L${CMAKE_INSTALL_FULL_LIBDIR}")
   if(UNIX OR MINGW)
     configure_file("${PROJECT_SOURCE_DIR}/${PROJECT_NAME}.pc.cmake.in" "${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc" @ONLY)
     install(FILES "${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc" DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
-    install(PROGRAMS "${PROJECT_SOURCE_DIR}/bin/coin-config" DESTINATION ${CMAKE_INSTALL_BINDIR})
-    configure_file("${PROJECT_SOURCE_DIR}/coin.cfg.cmake.in" "${CMAKE_BINARY_DIR}/${PROJECT_NAME_LOWER}-default.cfg" @ONLY)
-    install(FILES "${CMAKE_BINARY_DIR}/${PROJECT_NAME_LOWER}-default.cfg" DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/Coin/conf)
-  endif()
+  endif()
+endif()
+
+if(UNIX OR MINGW)
+  install(PROGRAMS "${PROJECT_SOURCE_DIR}/bin/coin-config" DESTINATION ${CMAKE_INSTALL_BINDIR})
+  configure_file("${PROJECT_SOURCE_DIR}/coin.cfg.cmake.in" "${CMAKE_BINARY_DIR}/${PROJECT_NAME_LOWER}-default.cfg" @ONLY)
+  install(FILES "${CMAKE_BINARY_DIR}/${PROJECT_NAME_LOWER}-default.cfg" DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/Coin/conf)
 endif()

 # Support find_package(Coin NO_MODULE).
--- coin.cfg.cmake.in.orig
+++ coin.cfg.cmake.in
@@ -4,17 +4,17 @@
 exec_prefix="@exec_prefix@"
 version="@COIN_VERSION@"
 cflags="@COIN_EXTRA_CFLAGS@"
-cppflags="@COIN_EXTRA_CPPFLAGS@"
+cppflags="-I@COIN_CONFIG_INCLUDEDIR@ @COIN_EXTRA_CPPFLAGS@"
 cxxflags="@COIN_EXTRA_CXXFLAGS@"
-ldflags="@COIN_EXTRA_LDFLAGS@"
-libs="@CMAKE_INSTALL_FULL_LIBDIR@"
+ldflags="@COIN_CONFIG_LDFLAGS@"
+libs="@COIN_EXTRA_LDFLAGS@"
 msvcrt="@COIN_MSVC_LIBC@"
 datadir="@CMAKE_INSTALL_FULL_DATADIR@"
-includedir="@CMAKE_INSTALL_FULL_INCLUDEDIR@"
-frameworkdir="@frameworkdir@"
+includedir="@COIN_CONFIG_INCLUDEDIR@"
+frameworkdir="@COIN_CONFIG_FRAMEWORKDIR@"
 htmldir="@CMAKE_INSTALL_FULL_DOCDIR@"
-compiler="@COIN_COMPILER@"
-objext="@OBJEXT@"
+compiler="@CMAKE_CXX_COMPILER@"
+objext="o"
 have_3ds_import=@FEAT_HAVE_3DS@
 have_vrml97=@FEAT_HAVE_VRML97@
 have_sound=@FEAT_HAVE_SOUND@
